### PR TITLE
Ticket-010 Clean-up for project

### PR DESCRIPTION
**Pull request by niculupu1**
1) Remove duplicated Networking Service
2) Add missing Protocols to all Services and layers
3) Insert all components into folders and move to according component
4) Remove lazy creation on modules and create proper inits

### CODE REVIEW

File: SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+class LaunchServiceImp: LaunchService {
     
-    private lazy var networkService = NetworkServiceImp()
-    private let url = "https://api.spacexdata.com/v3"
+    private var networkService: NewtorkService
+    private let baseURL = "https://api.spacexdata.com/v3"
     private let path: String = "/launches/"
     
+    init() {
+        networkService = NetworkServiceImp()
```
Comment (gcojocaru1) [inline SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift:17]: The purpose of initializing the networkService in init is to have it as an argument of the initializer.

Comment (gcojocaru1) [reply SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift:17]: ceva de genu `init(with networkService: NetworkService)`

Comment (niculupu1) [reply SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift:17]: Am in vedere cand chem initul in presenter. Trebuie sa creez o constanta de tip NetworkService() ?

File: SpaceX/LaunchesModule/LaunchDetails/Presenter/LaunchDetailsPresenterImp.swift [hunk 2]
```diff
@@ -11,13 +11,15 @@ class LaunchDetailsPresenterImp: LaunchDetailsPresenter {
     
     private let formater = ISO8601DateFormatter()
     private let dateFormat = DateFormatter()
-    private lazy var launchService = LaunchService()
+    private let launchService: LaunchService
     private var launchData: Flight?
+    private let networkService: NetworkService = NetworkServiceImp()
```
Comment (Vasile95) [inline SpaceX/LaunchesModule/LaunchDetails/Presenter/LaunchDetailsPresenterImp.swift:16]: network is part of LaunchServices and shouldn't be part of presenter/interactor so better to remove

File: SpaceX/NetworkService/NetworkServiceImp.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
+//
+//  NetworkService.swift
```
Comment (Vasile95) [inline SpaceX/NetworkService/NetworkServiceImp.swift:2]:
```suggestion
//  NetworkServiceImp.swift
```

File: SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
 import Foundation
 
-class LaunchService {
+class LaunchServiceImp: LaunchService {
     
-    private lazy var networkService = NetworkServiceImp()
-    private let url = "https://api.spacexdata.com/v3"
+    private var networkService: NetworkService
+    private let baseURL = "https://api.spacexdata.com/v3"
     private let path: String = "/launches/"
```
Comment (Vasile95) [inline SpaceX/LaunchesModule/LaunchServices/LaunchServiceImp.swift:14]: lets move this to an Enum with constants

enum {
      another enum for launch {
base url 
another enum with paths{}
}

.. same for other services
}

File: SpaceX/NetworkService/NetworkServiceImp.swift [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import UIKit
+
+//MARK: - Enums
+
+enum HttpMethod: String {
+    case get = "GET"
+    case put = "PUT"
+    case post = "POST"
+}
+
```
Comment (Vasile95) [inline SpaceX/NetworkService/NetworkServiceImp.swift:18]: lets move this to a separate file in constants folder. We will use this in multiple places

**Pull request merged**